v3/src/components/graph/components/graph-component.tsx

```diff
@@ -2,25 +2,33 @@ import {useDroppable} from '@dnd-kit/core'
 import {observer} from "mobx-react-lite"
 import React, {useEffect, useMemo, useRef, useState} from "react"
 import {useResizeDetector} from "react-resize-detector"
+import {useGraphController} from "../hooks/use-graph-controller"
 import {InstanceIdContext, useNextInstanceId} from "../../../hooks/use-instance-id-context"
 import {kTitleBarHeight} from "../graphing-types"
 import {AxisLayoutContext} from "../../axis/models/axis-layout-context"
-import {GraphLayout, GraphLayoutContext} from "../models/graph-layout"
+import {GraphLayoutContext, useGraphLayoutContext} from "../models/graph-layout"
 import {GraphModelContext, isGraphModel} from "../models/graph-model"
 import {Graph} from "./graph"
 import {ITileBaseProps} from '../../tiles/tile-base-props'
+import {GraphController} from "../models/graph-controller"
 
-export const GraphComponent = observer(({ tile }: ITileBaseProps) => {
+export const GraphComponent = observer(({tile}: ITileBaseProps) => {
   const graphModel = tile?.content
   if (!isGraphModel(graphModel)) return null
 
   const instanceId = useNextInstanceId("graph")
-  const layout = useMemo(() => new GraphLayout(), [])
+  const layout = useGraphLayoutContext()
```

Comment:
As discussed, I'll make this change in a separate PR to encapsulate all the new question marks

Reply:
I recommend changing this line back the way it was in _this_ PR. Then you can deal with the defaulting to `undefined` rather than a default context in the next PR. The two are logically separate and this line is logically part of this PR, not the next one.